### PR TITLE
Improved halide_popcount

### PR DESCRIPTION
**Pull request by Aelphy**
This is a slightly more efficient version of a popcount, which has as many iterations as many ones are in the binary representation.

### CODE REVIEW

File: src/CodeGen_C.cpp [hunk 1]
```diff
@@ -155,8 +155,9 @@ template<typename T>
 inline int halide_popcount(T a) {
     int bits_set = 0;
     while (a != 0) {
```
Comment (steven-johnson) [inline src/CodeGen_C.cpp:157]: FWIW, ~all modern compilers should have popcount as an intrinsic -- see `popcount64` in Util.cpp, maybe we should just adapt that here

**Comment by steven-johnson** [comment]
What's the story on this, ready to land?

File: src/CodeGen_C.cpp [hunk 2]
```diff
@@ -152,15 +152,39 @@ inline float float_from_bits(uint32_t bits) {
 }
 
 template<typename T>
-inline int halide_popcount(T a) {
+inline int halide_popcount_fallback(T a) {
```
Comment (vksnk) [inline src/CodeGen_C.cpp:155]: I would leave a link to the source for this algorithm for the reference (found this one https://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetKernighan, for example).

File: src/CodeGen_C.cpp [hunk 3]
```diff
@@ -152,15 +152,39 @@ inline float float_from_bits(uint32_t bits) {
...
     int bits_set = 0;
     while (a != 0) {
-        bits_set += a & 1;
-        a >>= 1;
+        bits_set += 1;
+        // NOTE(aelphy): remove least significant zeros and the first met one
```
Comment (vksnk) [inline src/CodeGen_C.cpp:159]: I would just make it a regular comment (i.e. remove NOTE(aelphy) part).

**Review by vksnk — APPROVED** [review]

**Comment by steven-johnson** [comment]
Where does this PR stand?

**Pull request merged**